client.go

```diff
@@ -180,31 +183,50 @@ func runStreamedCommand(img, cmd string, getError bool) error {
 // It returns the ID of the container spawned from the image.
 // Note well: the container is NOT deleted when the given command terminates.
 func runCommandInContainer(img string, cmd []string, dst io.Writer) (string, error) {
-	return startContainer(img, cmd, true, dst)
+	id, err := createContainer(img, cmd)
+	if err != nil {
+		log.Println(err)
+		return "", err
+	}
+
+	execute := true
+	for execute {
```

Comment:
To avoid infinite loops we could do something like

```
for i := 0; i < 256; i++ {
  // snip: startContainer
  switch err.(type) {
  case dockerError:
    // snip: the severe check
    if de.exitCode == zypperExitInfRestartNeeded {
      continue
    }
  default:
  }
  return id, err
}
return "", some_error_about_too_many_retries
```